examples/server/server.cpp

```diff
@@ -405,127 +510,625 @@ bool parse_str_to_bool(const std::string & s) {
+    } catch (const std::exception &e) {
+        SERVER_DEBUG("Error parsing numeric parameter: " << e.what());
+        set_error_response(res_for_error, std::string("Invalid numeric parameter: ") + e.what(), ErrorType::INVALID_REQUEST);
+        return false; // Indicate failure
     }
-    if (req.has_file("tinydiarize"))
-    {
-        params.tinydiarize = parse_str_to_bool(req.get_file_value("tinydiarize").content);
+
+    std::string debug_mode_str = get_param_value("debug_mode");
+    if (!debug_mode_str.empty()) params.debug_mode = parse_str_to_bool(debug_mode_str);
+    std::string translate_str = get_param_value("translate");
+    if (!translate_str.empty()) params.translate = parse_str_to_bool(translate_str);
+    std::string diarize_str = get_param_value("diarize");
+    if (!diarize_str.empty()) params.diarize = parse_str_to_bool(diarize_str);
+    std::string tinydiarize_str = get_param_value("tinydiarize");
+    if (!tinydiarize_str.empty()) params.tinydiarize = parse_str_to_bool(tinydiarize_str);
+    std::string split_on_word_str = get_param_value("split_on_word");
+    if (!split_on_word_str.empty()) params.split_on_word = parse_str_to_bool(split_on_word_str);
+    std::string no_timestamps_str = get_param_value("no_timestamps");
+    if (!no_timestamps_str.empty()) params.no_timestamps = parse_str_to_bool(no_timestamps_str);
+    std::string language_str = get_param_value("language");
+    if (!language_str.empty()) params.language = language_str;
+    std::string detect_language_str = get_param_value("detect_language");
+    if (!detect_language_str.empty()) params.detect_language = parse_str_to_bool(detect_language_str);
+    std::string prompt_str = get_param_value("prompt");
+    if (!prompt_str.empty()) params.prompt = prompt_str;
+    std::string response_format_str = get_param_value("response_format");
+    if (!response_format_str.empty()) params.response_format = response_format_str;
+    std::string suppress_non_speech_str = get_param_value("suppress_non_speech");
+    if (!suppress_non_speech_str.empty()) params.suppress_nst = parse_str_to_bool(suppress_non_speech_str);
+    std::string suppress_nst_str = get_param_value("suppress_nst");
+    if (!suppress_nst_str.empty()) params.suppress_nst = parse_str_to_bool(suppress_nst_str);
+    std::string no_context_str = get_param_value("no_context");
+    if (!no_context_str.empty()) params.no_context = parse_str_to_bool(no_context_str);
+    return true; // Indicate success
+}
+
+// --- Task Queue and Worker Thread for Concurrency ---
+struct TranscriptionTask {
+    int id;
+    std::shared_ptr<std::vector<float>> audio_data_pcmf32_ptr;
+    std::shared_ptr<std::vector<std::vector<float>>> audio_data_pcmf32s_ptr;
+    whisper_params params;
+    std::promise<json> result_promise;
+    std::string original_filename;
+    std::shared_ptr<std::atomic<bool>> connection_alive;
+    std::shared_ptr<std::atomic<bool>> cancel_flag;
+};
```

Comment:
`std::shared_ptr` should not be needed here.

`std::shared_ptr<std::atomic<bool>>` seems really suspicious.